src/components/flag/code/templates/FlagComponentTemplate.jsx

```diff
@@ -3,21 +3,12 @@ var classNames = require('classnames');
 
 module.exports = function(props) {
   var classes = {
-    'ui-component-flag': true,
-    'flag-left': props.left,
-    'flag-right': props.right,
-    'flag-top': props.top,
-    'flag-bottom': props.bottom,
-    'flag-primary': props.primary,
-    'flag-secondary': props.secondary,
-    'flag-success': props.success,
-    'flag-info': props.info,
-    'flag-warning': props.warning,
-    'flag-danger': props.danger,
-    'flag-sm': props.small,
-    'flag-md': props.medium,
-    'flag-lg': props.large
+    'component-flag': true,
   };
+  if(props.size) classes[props.size] = true;
+  if(props.purpose) classes[props.purpose] = true
+  if(props.position) classes[props.position] = true
```

Comment:
This might be controversial at HX, but it's generally not throughout communities that use curly braces. For instance, [Google's Java coding standards](https://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s4.1.1-braces-always-used) as well as [the O'Reilly book "Maintainable JavaScript"](https://books.google.com/books?id=J5RCV8yM7sEC&pg=PA201&lpg=PA201&dq=one+line+if+statements+bad+form&source=bl&ots=TEWvCwlRDr&sig=S9Mf1HAAsghvP6RWKFtUux3fFw4&hl=en&sa=X&ei=Qk9CVa-jCtCvogSAsIHYAQ&ved=0CE0Q6AEwBw#v=onepage&q=one%20line%20if%20statements%20bad%20form&f=false) both forbid this practice (as do the PEAR coding standards in PHP). Additionally, jQuery lists single line `if` statements [under bad examples](https://contribute.jquery.org/style-guide/js/#bad-examples).

Single line `if`s will always need to be refactored when new code is added, can be difficult to read/parse by future programmers, and can be a common place for new errors to occur. For instance, a major SSL/TLS bug was due to [a single line if statement that got fucked up](https://www.imperialviolet.org/2014/02/22/applebug.html).

Can someone on the other side of this argument give community or academic based reasoning why we should allow this shorthand method?